src/kbmod/search/KBMOSearch.cpp

```diff
@@ -235,60 +235,134 @@ void KBMOSearch::fillPsiAndPhiVects(const std::vector<RawImage>& psiImgs,
+                                                bool keep_no_data, bool all_stamps) {
+    std::vector<RawImage> stamps;
+    int num_times = stack.imgCount();
+    const trajectory& t = trj.get_const_trajectory();
+    for (int i = 0; i < num_times; ++i) {
+        if (all_stamps || trj.check_index_valid(i)) {
+            pixelPos pos = getTrajPos(t, i);
+            RawImage& img = stack.getSingleImage(i).getScience();
+            stamps.push_back(img.createStamp(pos.x, pos.y, radius, interpolate, keep_no_data));
+        }
+    }
+    return stamps;
+}
+
+inline std::vector<RawImage> KBMOSearch::scienceStampsForFilter(const TrajectoryResult& trj, int radius) {
+    return scienceStamps(trj, radius, false, true, false);
+}
+
+inline std::vector<RawImage> KBMOSearch::scienceStampsForViz(const TrajectoryResult& trj, int radius) {
+    return scienceStamps(trj, radius, true, false, true);
+}
+
+std::vector<RawImage> KBMOSearch::scienceStamps(trajectory& t, int radius) {
+    TrajectoryResult trj(t, stack.imgCount());
+    return scienceStampsForViz(trj, radius);
+}
+
+RawImage KBMOSearch::medianScienceStamp(const TrajectoryResult& trj, int radius, bool use_all) {
+    return createMedianImage(scienceStamps(trj, radius, false, true, use_all));
+}
+
+std::vector<RawImage> KBMOSearch::medianScienceStamps(const std::vector<TrajectoryResult>& t_array,
+                                                      int radius) {
+    const int num_results = t_array.size();
+    std::vector<RawImage> results(num_results);
+    omp_set_num_threads(16);
+
+#pragma omp parallel for
+    for (int s = 0; s < num_results; ++s) {
+        results[s] = medianScienceStamp(t_array[s], radius, true);
+    }
+    omp_set_num_threads(1);
+
+    return (results);
+}
+
+// To be deprecated in later PR.
 std::vector<RawImage> KBMOSearch::medianStamps(const std::vector<trajectory>& t_array,
                                                const std::vector<std::vector<int>>& goodIdx, int radius) {
-    int numResults = t_array.size();
-    int dim = radius * 2 + 1;
+    const int num_results = t_array.size();
+    std::vector<TrajectoryResult> arr;
+    for (int s = 0; s < num_results; ++s) {
+        TrajectoryResult trj(t_array[s], goodIdx[s]);
+        arr.push_back(trj);
+    }
 
-    std::vector<LayeredImage>& imgs = stack.getImages();
-    size_t N = imgs.size() / 2;
-    std::vector<RawImage> results(numResults);
+    return medianScienceStamps(arr, radius);
+}
+
+RawImage KBMOSearch::meanScienceStamp(const TrajectoryResult& trj, int radius, bool use_all) {
+    return createMeanImage(scienceStamps(trj, radius, false, true, use_all));
+}
+
+std::vector<RawImage> KBMOSearch::meanScienceStamps(const std::vector<TrajectoryResult>& t_array,
+                                                    int radius) {
+    const int num_results = t_array.size();
+    std::vector<RawImage> results(num_results);
     omp_set_num_threads(16);
```

Comment:
Just curious how you choose the number of threads for these calls `omp_set_num_threads(xx)`

Reply:
This has been carried over from the original code. It is unlikely to be optimal and we should do something more adaptive here.